packages/core/src/render3/instructions.ts

```diff
@@ -1465,6 +1464,31 @@ function instantiateAllDirectives(tView: TView, viewData: LViewData, previousOrP
   }
 }
 
+function invokeDirectivesHostBindings(
+    tView: TView, viewData: LViewData, previousOrParentTNode: TNode) {
+  const start = previousOrParentTNode.flags >> TNodeFlags.DirectiveStartingIndexShift;
+  const end = start + (previousOrParentTNode.flags & TNodeFlags.DirectiveCountMask);
+  const expando = tView.expandoInstructions !;
+  for (let i = start; i < end; i++) {
+    const def = tView.data[i] as DirectiveDef<any>;
+    const directive = viewData[i];
+    if (def.hostBindings) {
+      const before = expando.length;
+      setCurrentDirectiveDef(def);
+      def.hostBindings !(RenderFlags.Create, directive, previousOrParentTNode.index);
+      setCurrentDirectiveDef(null);
+      // `hostBindings` function may or may not contain `allocHostVars` call,
```

Comment:
Nice, I think the comment adds some good context. 

Nit: Maybe it would be even clearer to mention when `allocHostVars` call is omitted.

```suggestion
      // `hostBindings` function may or may not contain `allocHostVars` call 
      // (e.g. it may not if it only contains host listeners)
```